pkg/tsdb/elasticsearch/elasticsearch.go

```diff
@@ -78,6 +79,7 @@ func newInstanceSettings(httpClientProvider httpclient.Provider) datasource.Inst
 			return nil, fmt.Errorf("error reading settings: %w", err)
 		}
 		httpCliOpts, err := settings.HTTPClientOptions(ctx)
+		httpCliOpts.Middlewares = append(httpCliOpts.Middlewares, errorsource.Middleware("elasticsearch"))
```

Comment:
added a [New constructor](https://github.com/grafana/grafana-plugin-sdk-go/pull/795/files#diff-bebb0be07bf4d976fc20540020d140a95476dcd59431154dcc6aebcc1048343dR12) for http client.  so you don't have to explicitly add the middleware